hook-janitor/src/main.rs

```diff
@@ -0,0 +1,86 @@
+use axum::Router;
+use cleanup::{Cleaner, CleanerModeName};
+use config::Config;
+use envconfig::Envconfig;
+use eyre::Result;
+use futures::future::{select, Either};
+use kafka_producer::create_kafka_producer;
+use std::{str::FromStr, time::Duration};
+use tokio::sync::Semaphore;
+use webhooks::WebhookCleaner;
+
+use hook_common::metrics;
+
+mod cleanup;
+mod config;
+mod handlers;
+mod kafka_producer;
+mod webhooks;
+
+async fn listen(app: Router, bind: String) -> Result<()> {
+    let listener = tokio::net::TcpListener::bind(bind).await?;
+
+    axum::serve(listener, app).await?;
+
+    Ok(())
+}
+
+async fn cleanup_loop(cleaner: Box<dyn Cleaner>, interval_secs: u64) -> Result<()> {
+    let semaphore = Semaphore::new(1);
+    let mut interval = tokio::time::interval(Duration::from_secs(interval_secs));
+
+    loop {
+        let _permit = semaphore.acquire().await;
+        interval.tick().await;
+        cleaner.cleanup().await;
```

Comment:
Should we spawn a task here? Although I don't know if we need concurrency; in fact multiple cleaners running at the same time could be a problem.

Reply:
Yeah, we want to block and complete. Let's say we run every 60 seconds and for some reason this starts taking 70 seconds, I don't want them to overlap.